pkgs/applications/misc/mangal/default.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, buildGoModule, fetchFromGitHub, installShellFiles, go_1_18, lua }:
+
+buildGoModule rec {
+  pname = "mangal";
+  version = "3.0.3";
+
+  src = fetchFromGitHub {
+    owner = "metafates";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-+U3KDIfpHWzWJrw0DoF1wySwLfFNvknPBUEhl6IowFo=";
+  };
+
+  vendorSha256 = "sha256-PrslSjWkiLJiHnJFuIz2+RU7S446ahcf31nQ6GYfAhU=";
+
+  ldflags = [ "-s" "-w" ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
+    installShellCompletion --cmd mangal \
+      --bash <($out/bin/mangal completion bash) \
+      --zsh <($out/bin/mangal completion zsh) \
+      --fish <($out/bin/mangal completion fish)
+  '';
+
+  doCheck = false; # test fail because of sandbox
+
+  meta = with lib; {
+    description = "A fancy CLI app written in Go which scrapes, downloads and packs manga into different formats";
+    homepage = "https://github.com/metafates/mangal";
+    license = licenses.mit;
+    maintainers = [ maintainers.bertof ];
+  };
```

Comment:
Missing `platforms`.

Reply:
Sorry, I messed up, now I checked that if not set it will be set by default by `buildGoModule`, so you can remove it unless there is a reason to specify custom platforms.

Reply:
No problem. I updated the derivation removing the entry.